flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/CopyOnWriteStateTable.java

```diff
@@ -287,6 +289,14 @@ public S get(K key, N namespace) {
 	}
 
 	@Override
+	public Stream<K> getKeys(N namespace) {
+		return Arrays.stream(primaryTable)
```

Comment:
This hash table applies incremental rehashing. This method must also consider the data stored ``incrementalRehashTable`` if it isn't ``null`` or we can miss some data.

Reply:
also added larger test that covers this case